### PR TITLE
feat(users): add follow/unfollow API

### PR DESCRIPTION
**Pull request by max-wittig**

### CODE REVIEW

File: docs/gl_objects/users.rst [hunk 1]
```diff
@@ -84,6 +89,15 @@ Delete an external identity by provider name::
 
     user.identityproviders.delete('oauth2_generic')
 
+Get the followers of a user
+
+    user.followers_users.list()
+
+Get the followings of a user
+
+    user.following_users.list()
```
Comment (max-wittig) [inline docs/gl_objects/users.rst:98]: @nejch I couldn't use `following` and `followers` directly, as it has the same name as it's user attribute.

Comment (max-wittig) [reply docs/gl_objects/users.rst:98]: What do you think about this naming?

File: gitlab/tests/objects/test_users.py [hunk 2]
```diff
@@ -135,3 +201,17 @@ def test_user_activate_deactivate(user, resp_activate):
 
 def test_delete_user_identity(user, resp_delete_user_identity):
     user.identityproviders.delete("test_provider")
+
+
+def test_user_follow_unfollow(user, resp_follow_unfollow):
+    user.follow()
```
Comment (max-wittig) [inline gitlab/tests/objects/test_users.py:207]: sadly in the test case only the `dict` is returned, so I cannot really assert the attributes of the `User` object.

Any ideas how to convert this to a `User` in the tests? It does work normally

**Review by nejch — CHANGES_REQUESTED** [review]
Cool! It all looks good to me, just that little docstring return (unless you want to return a User, see other thread).

File: gitlab/tests/objects/test_users.py (hunk 2 above)
Comment (max-wittig) [reply gitlab/tests/objects/test_users.py:207]: Okay that makes sense. Let's keep the conventions then.

File: docs/gl_objects/users.rst (hunk 1 above)
Comment (max-wittig) [reply docs/gl_objects/users.rst:98]: Then let's keep the clearer name I guess 😄

**Review by nejch — APPROVED** [review]

**Pull request merged**